primitives/api/proc-macro/src/decl_runtime_apis.rs

```diff
@@ -452,16 +432,38 @@ fn generate_runtime_decls(decls: &[ItemTrait]) -> Result<TokenStream> {
 
 		let call_api_at_calls = generate_call_api_at_calls(&decl)?;
 
+		let trait_api_version = get_api_version(&found_attributes).unwrap_or(1);
+
 		// Remove methods that have the `changed_in` attribute as they are not required for the
 		// runtime anymore.
+		// Generate default implementations for staging methods.
 		decl.items = decl
 			.items
 			.iter_mut()
 			.filter_map(|i| match i {
 				TraitItem::Method(ref mut method) => {
-					if remove_supported_attributes(&mut method.attrs)
-						.contains_key(CHANGED_IN_ATTRIBUTE)
-					{
+					let method_attrs = remove_supported_attributes(&mut method.attrs);
+
+					if let Some(version_attribute) = method_attrs.get(API_VERSION_ATTRIBUTE) {
```

Comment:
Each staging method has got a default implementation in the trait calling `unimplemented!`. I probably need to do this in `impl_runtime_apis!` somehow so that proper errors can be generated if a method is not implemented.